file.cc

```diff
@@ -269,102 +164,43 @@ bool Class::unmap( uchar * address )
 }
 
 
-void Class::seekCur( qint64 offset ) 
-{
-  if ( writeBuffer )
-    flushWriteBuffer();
-
-  if( !f.seek( f.pos() + offset ) )
-    throw exSeekError();
-}
-
-void Class::seekEnd( qint64 offset ) 
+void Class::seekEnd()
 {
-  if ( writeBuffer )
-    flushWriteBuffer();
-
-  if( !f.seek( f.size() + offset ) )
+  if ( !f.seek( f.size()) )
     throw exSeekError();
 }
 
-void Class::rewind() 
+void Class::rewind()
 {
   seek( 0 );
 }
 
-qint64 Class::tell() 
+qint64 Class::tell()
 {
-  qint64 result = f.pos();
-
-  if ( result == -1 )
-    throw exSeekError();
-
-  if ( writeBuffer )
-    result += ( WriteBufferSize - writeBufferLeft );
-
-  return result;
+  return f.pos();
 }
 
-bool Class::eof() 
+bool Class::eof()
 {
-  if ( writeBuffer )
-    flushWriteBuffer();
-
   return f.atEnd();
 }
 
-QFile & Class::file() 
+QFile & Class::file()
 {
-  flushWriteBuffer();
+  f.flush();
 
```

Comment:
I think flushWriteBuffer has not ensured a flush() will be called.
`f.flush()` should be fine.

Reply:
Removed.